### ISSUE TITLE
array.allocated is giving the wrong numbers

### ISSUE THREAD

**Issue by pavanky** [arrayfire/arrayfire#1862]
code to reproduce:

```
array a = af::randu(5);
std::cout << a.allocated() << std::endl; // prints 20
af::printMemInfo("memory", 0); // displays 1kb used 
```

**Comment by pavanky** [comment]
This is working as per the original expectation. We should probably figure out if we want to change the behavior to specify the memory use by memory manager or the original number of elements * sizeof(datatype)

**Comment by pavanky** [comment]
@umar456

**Comment by pavanky** [comment]
Fixed in hotfixes-3.5.1
